Source/Core/DolphinQt2/QtUtils/SignalDaemon.cpp

```diff
@@ -0,0 +1,34 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/QtUtils/SignalDaemon.h"
+
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <QSocketNotifier>
+
+int SignalDaemon::s_sigterm_fd[2];
+
+SignalDaemon::SignalDaemon(QObject* parent) : QObject(parent)
+{
+  if (socketpair(AF_UNIX, SOCK_STREAM, 0, s_sigterm_fd))
+    qFatal("Couldn't create TERM socketpair");
+
+  m_term = new QSocketNotifier(s_sigterm_fd[1], QSocketNotifier::Read, this);
+  connect(m_term, &QSocketNotifier::activated, this, &SignalDaemon::InterruptReceived);
+}
+
+void SignalDaemon::HandleInterrupt(int)
+{
+  const auto message =
+      QObject::tr("\nA signal was received. A second signal will force Dolphin to stop.\n")
```

Comment:
This mustn't be localised otherwise you'll run into the same signal safety problem